### DISCUSSION TITLE
Model import order with STI

### DISCUSSION THREAD

**Discussion by Sayanel34** [CodeDredd/pinia-orm#1703] [Q&A]
Hi !
I have a very precise way to use models, first with vuex-orm and now with pinia-orm.
I have a model called "ApiPlatformModel" which inherit from pinia-orm's Model.
This ApiPlatformModel is here to set a base to all models i'll use after that.

Now let's say i have a "Person" model, which inherit ApiPlatformModel but without the base entity, cause it's just a base model.
Everything's okay now, it's all perfect.

Now i have a "PhysicalPerson" model which inherit Person model, this time with baseEntity etc... Still no problem, works like a charm.

My real problem is here : i compile this package with my models cause i'll use them in different apps. Compilation seems okay, but now i want to use them in a Nuxt static application.
And when i want to use them, i'm getting an error : Class extends value undefined is not a constructor or null.
It seems that it's trying to use a child model before fetching the parent.
I had no problem before because we could chose the import order and reference to store, but now everything is auto imported through the extending of pinia-orm's Model.

I don't know if it's all clear, let me know if not. If anyone has something in mind i'll be glad to test it.

Regards.

**Comment by CodeDredd** [comment]
Hey @Sayanel34 ,
thanks for your post. If you could provide a working reproduciton either with my playground or with stackblitz it would be great. 🙏  As an open source developer doing this in my free spare time i don't have time to make for every ticket a reproduction.
You would help me with that a lot! ❤️

**Reply by Sayanel34** [reply]
Hi, sorry for the delay.
It's very hard to know what is going on, cause after changing a lot of things, any STI inheritance cause the problem... randomly.
Cache can be a problem too, but not always.

We're using Nuxt 3 and are facing this issue : https://github.com/CodeDredd/pinia-orm/issues/1610 with a ManyToMany relationship.
I don't know if it's related, but those two problems are quite hectic.
I'll try to make a working reproduction, but it will probably be hard for the moment.

**Reply by Sayanel34** [reply]
Well i think i could make a PR to fix my issue, it seems you're referencing children before parents and it's quite not working with Nuxt.
